### DISCUSSION TITLE
Provide a way to access disconnect reason & connect error in FlaskIOTestClient

### DISCUSSION THREAD

**Discussion by wbobeirne** [miguelgrinberg/Flask-SocketIO#1599] [Q&A]
**Is your feature request related to a problem? Please describe.**

Our codebase has a bunch of specific disconnect reasons that we'd like to write tests for (Too many connections, bad authentication, etc.) Right now we write the tests just checking if `is_connected() == False`, however it would be nice if we could test which error came back.

**Describe the solution you'd like**

Something to the tune of:
```py
assert(sio.is_connected() == False)
assert(sio.get_connect_error().message == 'MY_CUSTOM_ERROR_CODE')
assert(sio.get_disconnect_reason() == 'io server disconnect')
```

They could also raise if you're still connected ala `get_received`

**Describe alternatives you've considered**

I could mock the functions to check when they raise or disconnect, but this ties my tests really tightly to the server implementation. It would be nice if I could do some refactoring without having to update my test mocks.

**Comment by miguelgrinberg** [comment]
How are you planning to set the error code/message and disconnect reason from the server side?

**Comment by wbobeirne** [comment]
Either by calling the `disconnect` function manually, or raising a `ConnectionRefusedError`.

**Comment by miguelgrinberg** [comment]
The disconnect function does not have a way to provide details. The `ConnectionRefusedError` does have a custom payload, but there isn't anything such as a disconnect reason, or an error code, so none of the ways you envision this to work are really doable.

I think it might be possible to access the value(s) that you include when you raise the exception, but that is far as this feature can go, it can't be mode more specific than that because the Socket.IO protocol does not have any of the things you want to check for.
